twilight-gateway/src/shard.rs

```diff
@@ -1084,9 +1084,9 @@ impl Shard {
         future::reconnect_delay(reconnect_attempts).await;
 
         let maybe_gateway_url = self
-            .config
-            .gateway_url()
-            .or(self.resume_gateway_url.as_deref());
+            .resume_gateway_url
+            .as_deref()
+            .or_else(|| self.config.proxy_url());
```

Comment:
Is this change correct, we should always use the proxy url if it is there not only if the resume url is not.

Reply:
The change corresponds to the second item in the OP bullet list:
> Resumes to resume_gateway_url instead of proxy_url as it's less surprising, is the same behaviour as in twilight-gateway 0.14, and proxy gateways construct their own Ready event either way.